ts/src/bitrue.ts

```diff
@@ -741,50 +1004,159 @@ export default class bitrue extends Exchange {
          * @method
          * @name bitrue#fetchBalance
          * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @see https://github.com/Bitrue-exchange/Spot-official-api-docs#account-information-user_data
+         * @see https://www.bitrue.com/api-docs#account-information-v2-user_data-hmac-sha256
+         * @see https://www.bitrue.com/api_docs_includes_file/delivery.html#account-information-v2-user_data-hmac-sha256
          * @param {object} [params] extra parameters specific to the bitrue api endpoint
+         * @param {string} [params.type] 'future', 'delivery', 'spot', 'swap'
+         * @param {string} [params.subType] 'linear', 'inverse'
          * @returns {object} a [balance structure]{@link https://docs.ccxt.com/#/?id=balance-structure}
          */
         await this.loadMarkets ();
-        const response = await this.v1PrivateGetAccount (params);
-        //
-        //     {
-        //         "makerCommission":0,
-        //         "takerCommission":0,
-        //         "buyerCommission":0,
-        //         "sellerCommission":0,
-        //         "updateTime":null,
-        //         "balances":[
-        //             {"asset":"sbr","free":"0","locked":"0"},
-        //             {"asset":"ksm","free":"0","locked":"0"},
-        //             {"asset":"neo3s","free":"0","locked":"0"},
-        //         ],
-        //         "canTrade":false,
-        //         "canWithdraw":false,
-        //         "canDeposit":false
-        //     }
-        //
-        return this.parseBalance (response);
+        const defaultType = this.safeString2 (this.options, 'fetchBalance', 'defaultType', 'spot');
+        const type = this.safeString (params, 'type', defaultType);
+        params = this.omit (params, 'type');
+        let subType = undefined;
```

Comment:
for instance here makes total sense to use handleMarketTypeAndParams instead of doing these manual checks